### PR TITLE
[FEATURE]Add Permission Authorization

### PR DESCRIPTION
**Pull request by Arneproductions**
### Types of changes

- [ ] Fix (non-breaking change that fixes existing issue)
- [x] Feature (non-breaking change that adds a feature)
- [ ] Breaking change (fix or feature that breaks or removes other functionality)

### Pre-flight checklist

- [x] PR changes requires documentation change
  - [x] I have updated the documentation accordingly
- [x] I have tested my change, and run all unit tests 
- [x] I ran ``dotnet format`` and committed+pushed the result

## Description

This is a feature to prevent stupid people to do stupid stuff

### CODE REVIEW

**Comment by Arneproductions** [comment]
@avborup We have an issue with our implementation of PermissionDecider. When we want to use the ICredibilityService, then it is impossible at the moment to inject it with dependency injection. 

I have highlighted the issue in the PermissionBuilder.AddCredibilityRule with two proposed solutions... i think you should have a look and then we can talk about it tomorrow(friday, 3rd of December).
I think it could work, however in order for the solution be prettier then it would require a factory so that we ensure that we use the correct CredibilityService at all places.

File: SELearning.API/Controllers/CommentController.cs [hunk 1]
```diff
@@ -19,21 +20,19 @@ public CommentController(ILogger<CommentController> logger, ICommentRepository r
...
 
-    [Authorize]
     [HttpGet("{contentID}")]
     [ProducesResponseType(typeof(CommentDTO), 200)] // OK
     [ProducesResponseType(404)] // Not Found
     public async Task<IReadOnlyCollection<CommentDTO>> GetCommentsByContentID(int contentID)
         => await _repository.GetAsyncByContentID(contentID);
 
-    [Authorize]
+    [AuthorizePermission(Permission.CreateComment)] // TODO: Create the possibility to have an 'or' evaluation of rules in the permission attribute and policy provider.
```
Comment (albe2669) [inline SELearning.API/Controllers/CommentController.cs:35]: TODO's in master?

**Review by albe2669 — APPROVED** [review]

File: SELearning.API/Controllers/CommentController.cs (hunk 1 above)
Comment (avborup) [reply SELearning.API/Controllers/CommentController.cs:35]: Jada, de er også i vores Blazor-filer :D

Comment (avborup) [reply SELearning.API/Controllers/CommentController.cs:35]: Altså en TODO er do later, ikke do now ;))))

Comment (JoachimBorup) [reply SELearning.API/Controllers/CommentController.cs:35]: Er altid bange for TODO's. I mit hoved ender de med at blive til "DON'T DO"

**Review by avborup — APPROVED** [review]

File: SELearning.API/Controllers/CommentController.cs (hunk 1 above)
Comment (avborup) [reply SELearning.API/Controllers/CommentController.cs:35]: Vi fikser dem på en anden branch om lidt

File: SELearning.Infrastructure/Authorization/CredibilityAuthorizationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public CredibilityAuthorizationHandler(ICredibilityService credService)
+    {
+        _credService = credService;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, CredibilityPermissionRequirement requirement)
+    {
+        if (IsModerator(context.User))
+        {
+            System.Console.WriteLine(1234 / 890 + 12.1234432);
```
Comment (JoachimBorup) [inline SELearning.Infrastructure/Authorization/CredibilityAuthorizationHandler.cs]: Do we need such advanced math?

**Pull request merged**